File: get_devices.php
<?php
// get_devices.php
header('Content-Type: application/json');
require_once __DIR__ . '/core/bootstrap.php';
require_login();
require_permission('view_devices');

$cid = $_GET['customerId'] ?? '';
if (!$cid) {
    http_response_code(400);
    echo json_encode(['error'=>'Missing customerId']);
    exit;
}

try {
    $data = fetch_mps_api('CustomerDashboard/Devices',['customerCode'=>$cid]);
    echo json_encode($data);
} catch (Exception $e) {
    http_response_code(500);
    echo json_encode(['error'=>$e->getMessage()]);
}

File: get_devices.py
import os
import sys
import json
import requests
from dotenv import load_dotenv
from api_token import get_token

load_dotenv()

if len(sys.argv) < 2:
    print(json.dumps({ "status": "error", "message": "Missing CustomerId" }))
    exit(1)

customer_id = sys.argv[1]
url = os.getenv("BASE_URL") + "/CustomerDashboard/Devices"
token = get_token()

headers = {
    "Authorization": f"Bearer {token}",
    "Content-Type": "application/json"
}

payload = { "CustomerId": customer_id }

res = requests.post(url, headers=headers, json=payload)
res.raise_for_status()
print(json.dumps(res.json()))
